### PR TITLE
add  invoker to detect  fingerprint hardware is enabled or not 

### PR DESCRIPTION
**Pull request by rickyazhari**

### CODE REVIEW

**Comment by rickyazhari** [comment]
I signed it!

File: packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java [hunk 1]
```diff
@@ -447,14 +446,18 @@ public void onPathReady(String path) {
...
   private void handleImageResult(String path) {
-    if (pendingResult != null) {
-      Double maxWidth = methodCall.argument("maxWidth");
-      Double maxHeight = methodCall.argument("maxHeight");
-
-      String finalImagePath = imageResizer.resizeImageIfNeeded(path, maxWidth, maxHeight);
-      finishWithSuccess(finalImagePath);
-    } else {
-      throw new IllegalStateException("Received image from picker that was not requested");
+    try{
```
Comment (bkonyi) [inline packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java]: This code doesn't look relevant to fingerprint functionality. Is this change here intentionally?

File: packages/image_picker/lib/image_picker.dart [hunk 2]
```diff
@@ -52,7 +52,7 @@ class ImagePicker {
         'maxHeight': maxHeight,
       },
     );
-
+    print("image picker : "+path);
```
Comment (bkonyi) [inline packages/image_picker/lib/image_picker.dart]: This code doesn't look relevant to fingerprint functionality. Is this change here intentionally?

File: packages/local_auth/android/src/main/java/io/flutter/plugins/localauth/AuthenticationHelper.java [hunk 3]
```diff
@@ -92,8 +92,18 @@
     this.fingerprintManager = FingerprintManagerCompat.from(activity);
   }
 
+
+  boolean isFingerPrintAvailable(){
+    if(android.os.Build.VERSION.SDK_INT >= 23){
```
Comment (bkonyi) [inline packages/local_auth/android/src/main/java/io/flutter/plugins/localauth/AuthenticationHelper.java]: This could be combined into a single return 
```
return ((android.os.Build.Version.SDK_INT >= 23) && 
        fingerprintManager.isHardwareDetected());
```

File: packages/local_auth/android/src/main/java/io/flutter/plugins/localauth/AuthenticationHelper.java [hunk 4]
```diff
@@ -92,8 +92,18 @@
     this.fingerprintManager = FingerprintManagerCompat.from(activity);
   }
 
+
+  boolean isFingerPrintAvailable(){
```
Comment (bkonyi) [inline packages/local_auth/android/src/main/java/io/flutter/plugins/localauth/AuthenticationHelper.java]: Nit: I'd change this to `isFingerprintReaderAvailable`

File: packages/local_auth/ios/Classes/LocalAuthPlugin.m [hunk 5]
```diff
@@ -62,16 +68,26 @@ - (void)alertMessage:(NSString *)message
                                                                                    completion:nil];
 }
 
+- (BOOL)isFingerprintAvailable{
```
Comment (bkonyi) [inline packages/local_auth/ios/Classes/LocalAuthPlugin.m]: Nit: same naming recommendation as above. `isFingerprintReaderAvailable`

File: packages/local_auth/ios/Classes/LocalAuthPlugin.m [hunk 6]
```diff
@@ -62,16 +68,26 @@ - (void)alertMessage:(NSString *)message
...
                  withFlutterResult:(FlutterResult)result {
   LAContext *context = [[LAContext alloc] init];
   NSError *authError = nil;
   lastCallArgs = nil;
   lastResult = nil;
   context.localizedFallbackTitle = @"";
-
-  if ([context canEvaluatePolicy:LAPolicyDeviceOwnerAuthenticationWithBiometrics
-                           error:&authError]) {
+  BOOL isAvailable = [self isFingerprintAvailable];
```
Comment (bkonyi) [inline packages/local_auth/ios/Classes/LocalAuthPlugin.m]: I don't think we need to actually create a local variable for this just to check it in the conditional.

File: packages/local_auth/lib/local_auth.dart [hunk 7]
```diff
@@ -73,4 +73,8 @@ class LocalAuthentication {
     }
     return await _channel.invokeMethod('authenticateWithBiometrics', args);
   }
+
+  Future<bool> isFingerprintAVailable() async{
```
Comment (bkonyi) [inline packages/local_auth/lib/local_auth.dart]:
```dart
Future<bool> isFingerprintReaderAvailable() async => 
    await _channel.invokeMethod('isFingerprintReaderAvailable');
```

File: packages/local_auth/ios/Classes/LocalAuthPlugin.m [hunk 8]
```diff
@@ -21,6 +21,12 @@ + (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
 - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result {
   if ([@"authenticateWithBiometrics" isEqualToString:call.method]) {
     [self authenticateWithBiometrics:call.arguments withFlutterResult:result];
+  }else if([@"isFingerPrintAvailable" isEqualToString:call.method]){
+    if([self isFingerprintAvailable]){
```
Comment (bkonyi) [inline packages/local_auth/ios/Classes/LocalAuthPlugin.m]: This entire block can be reduced to `result([self isFingerprintAvailable]);`

File: packages/local_auth/android/src/main/java/io/flutter/plugins/localauth/LocalAuthPlugin.java [hunk 9]
```diff
@@ -72,6 +72,15 @@ public void onError(String code, String error) {
...
       authenticationHelper.authenticate();
+    } else if(call.method.equals("isFingerPrintAvailable")){
+      Activity activity = registrar.activity();
+      AuthenticationHelper authenticationHelper = new AuthenticationHelper(
+        activity,
+        call,
+        null
+      );
+      boolean isAvailable = authenticationHelper.isFingerPrintAvailable();
+      result.success(isAvailable);
```
Comment (bkonyi) [inline packages/local_auth/android/src/main/java/io/flutter/plugins/localauth/LocalAuthPlugin.java]: Should just be `result.success(authenticationHelper.isFingerPrintAvailable());`

**Comment by bkonyi** [comment]
Closing due to inactivity. Feel free to reopen if we want to move forward with this.
